lib/chef/resource/locale.rb

```diff
@@ -0,0 +1,86 @@
+#
+# Copyright:: 2011-2016, Heavy Water Software Inc.
+# Copyright:: 2016-2018, Chef Software Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific slanguage governing permissions and
+# limitations under the License.
+#
+
+require "chef/resource"
+
+class Chef
+  class Resource
+    class Locale < Chef::Resource
+      resource_name :locale
+
+      description "Use the locale resource to set the system's locale"
+      introduced "14.5"
+
+      property :lang, String, default: "en_US.utf8"
```

Comment:
You'll want to add description fields here as well that explains what these new properties do. You can check out the other resources in this same dir for an example.